app/controllers/rooms_controller.rb

```diff
@@ -199,6 +230,74 @@ def send_scores
       ::WebsocketRails[:"room#{params[:id]}"].trigger "team_b_score", game_logic.showable_team_b_score
     end
 
+    # @param room [Room]
+    def save_history(room)
+      room_players = @room.room_players
+
+      # @type [Array<RoomPlayer]
+      team_a_players = room_players.select {|room_player| room_player.team == PlayersController::TEAM_A_ID}
+      # @type [Array<RoomPlayer]
+      team_b_players = room_players.select {|room_player| room_player.team == PlayersController::TEAM_B_ID}
+
+      # We should batch this but ActiveRecord makes it a pain in the ass and I don't care that much
+      # @type [Array<GameHistory>]
+      histories = []
+      if room.team_a_score > room.team_b_score
+        # Yay for team A
+        team_a_players.each do |room_player|
+          history = new_game_history(room, room_player, true)
+          histories << history
+          history.save
+        end
+        team_b_players.each do |room_player|
+          history = new_game_history(room, room_player, false)
+          histories << history
+          history.save
+        end
+      else
+        team_a_players.each do |room_player|
+          history = new_game_history(room, room_player, false)
+          histories << history
+          history.save
+        end
+        # All glory to team B
+        team_b_players.each do |room_player|
+          history = new_game_history(room, room_player, true)
+          histories << history
+          history.save
```

Comment:
Yeah, fair enough. I'd originally had it working differently and this is an artifact of that.